### PR TITLE
 Add Very Big Enum with all commands to protobuf

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
there are 4 changes:
1. added more enum vars
2. renames like `Hvals` to `HVals`
3. renames like `HashDel` to `HDel`
4. renames of the corresponding functions

All renames (except `HashSet`) were done with `Ctrl+Shift+F` magic

### CODE REVIEW

File: glide-core/src/protobuf/redis_request.proto [hunk 1]
```diff
@@ -38,118 +38,553 @@ message Routes {
...
+    AclList                        = 1106;
+    AclLoad                        = 1107;
+    AclLog                         = 1108;
+    AclSave                        = 1109;
+    AclSetSser                     = 1110;
+    AclUsers                       = 1111;
+    AclWhoami                      = 1112;
+    BgRewriteAof                   = 1113;
+    BgSave                         = 1114;
+    Command_                       = 1115; // TODO renaming to `Command` causes protobuf error
```
Comment (Yury-Fridlyand) [inline glide-core/src/protobuf/redis_request.proto:268]: If I rename this to `Command`, protobuf returns
```
  protobuf/redis_request.proto:268:5: "Command" is already defined in "redis_request".
  protobuf/redis_request.proto:268:5: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, "Command" must be unique within "redis_request", not just within "RequestType".
```
It conflicts with `Command` on line 590. Which one to rename?

File: glide-core/src/protobuf/redis_request.proto [hunk 2]
```diff
@@ -38,118 +38,553 @@ message Routes {
...
+    Lcs                            = 1512;
+    LcsLen                         = 1513;
+    LcsIdx                         = 1514;
+    MGet                           = 1515;
+    MSet                           = 1516;
+    MSetNx                         = 1517;
+    PSetEx                         = 1518; // deprecated in 2.6.12
+    SetString                      = 1519; // Set
+    SetEx                          = 1520; // deprecated in 2.6.12
+    SetNx                          = 1521; // deprecated in 2.6.12
```
Comment (Yury-Fridlyand) [inline glide-core/src/protobuf/redis_request.proto:422]: keep it there to avoid blanks

File: glide-core/src/protobuf/redis_request.proto [hunk 3]
```diff
@@ -38,118 +38,553 @@ message Routes {
...
-    LInsert = 101;
-    RPushX = 102;
-    LPushX = 103;
-    ZMScore = 104;
-    ZDiff = 105;
-    ZDiffStore = 106;
-    SetRange = 107;
-    ZRemRangeByLex = 108;
-    ZLexCount = 109;
+    GetString                      = 1504; // Get
```
Comment (Yury-Fridlyand) [inline glide-core/src/protobuf/redis_request.proto:405]: Will rename to `Get` and `Set` if no objections

File: csharp/lib/AsyncClient.cs [hunk 4]
```diff
@@ -127,107 +127,528 @@ private void SuccessCallback(ulong index, IntPtr str)
...
+        FlushDb = 1129,
+        Info = 1130,
+        LastSave = 1131,
+        LatencyDoctor = 1132,
+        LatencyGraph = 1133,
+        LatencyHistogram = 1134,
+        LatencyHistory = 1135,
+        LatencyLatest = 1136,
+        LatencyReset = 1137,
+        Lolwut = 1138,
```
Comment (jonathanl-bq) [inline csharp/lib/AsyncClient.cs:370]: I can't believe this is actually real... https://redis.io/docs/latest/commands/lolwut/

File: glide-core/src/protobuf/redis_request.proto [hunk 5]
```diff
@@ -38,118 +38,553 @@ message Routes {
...
+    ClientSetInfo                  = 314;
+    ClientSetName                  = 315;
+    ClientTracking                 = 316;
+    ClientTrackingInfo             = 317;
+    ClientUnblock                  = 318;
+    ClientUnpause                  = 319;
+    Echo                           = 320;
+    Hello                          = 321;
+    Ping                           = 322;
+    Quit                           = 323; // deprecated in 7.2.0
```
Comment (jonathanl-bq) [inline glide-core/src/protobuf/redis_request.proto:112]: Do we really need to add the deprecated variants? I guess it doesn't hurt, but I wonder what the AWS side will want.

File: csharp/lib/AsyncClient.cs (hunk 4 above)
Comment (Yury-Fridlyand) [reply csharp/lib/AsyncClient.cs:370]: I tested it with custom command and looking forward adding implementation of that!

File: glide-core/src/protobuf/redis_request.proto (hunk 3 above)
Comment (jonathanl-bq) [reply glide-core/src/protobuf/redis_request.proto:405]: I like the idea. It keeps things more consistent IMO.

File: glide-core/src/protobuf/redis_request.proto (hunk 1 above)
Comment (jonathanl-bq) [reply glide-core/src/protobuf/redis_request.proto:268]: It would probably impact less code to rename the Command here instead of the one on line 590. However, I think the "right" way to do this is to rename the one at line 590 in a separate PR.

File: glide-core/src/request_type.rs [hunk 6]
```diff
@@ -9,115 +9,528 @@ use crate::redis_request::RequestType as ProtobufRequestType;
...
+    XReadGroup = 1418,
+    XRevRange = 1419,
+    XSetId = 1420,
+    XTrim = 1421,
+
+    //// String commands : https://redis.io/commands/?group=string
+    Append = 1501,
+    Decr = 1502,
+    DecrBy = 1503,
+    /// Type of a get string request.
```
Comment (jonathanl-bq) [inline glide-core/src/request_type.rs:361]: This comment feels kind of random and out of place.

File: glide-core/src/request_type.rs [hunk 7]
```diff
@@ -157,22 +570,22 @@ impl From<::protobuf::EnumOrUnknown<ProtobufRequestType>> for RequestType {
             ProtobufRequestType::ClientUnblock => RequestType::ClientUnblock,
             ProtobufRequestType::ClientUnpause => RequestType::ClientUnpause,
             ProtobufRequestType::Expire => RequestType::Expire,
-            ProtobufRequestType::HashSet => RequestType::HashSet,
-            ProtobufRequestType::HashGet => RequestType::HashGet,
-            ProtobufRequestType::HashDel => RequestType::HashDel,
-            ProtobufRequestType::HashExists => RequestType::HashExists,
+            ProtobufRequestType::HSet => RequestType::HSet,
```
Comment (jonathanl-bq) [inline glide-core/src/request_type.rs:573]: I like these changes but we might need to split them into a separate PR due to how large this PR already is. Not a strong opinion though, and not a blocker.

File: glide-core/src/request_type.rs [hunk 8]
```diff
@@ -329,33 +743,34 @@ impl RequestType {
...
             RequestType::LInsert => Some(cmd("LINSERT")),
-            RequestType::Spop => Some(cmd("SPOP")),
+            RequestType::SPop => Some(cmd("SPOP")),
             RequestType::ZMScore => Some(cmd("ZMSCORE")),
             RequestType::ZDiff => Some(cmd("ZDIFF")),
             RequestType::ZDiffStore => Some(cmd("ZDIFFSTORE")),
             RequestType::SetRange => Some(cmd("SETRANGE")),
             RequestType::ZRemRangeByLex => Some(cmd("ZREMRANGEBYLEX")),
             RequestType::ZLexCount => Some(cmd("ZLEXCOUNT")),
+            _ => todo!(),
```
Comment (jonathanl-bq) [inline glide-core/src/request_type.rs:773]: Are there actually more commands we're missing here? Or is this just a catch-all case for future added commands?

File: java/client/src/main/java/glide/api/models/commands/SetOptions.java [hunk 9]
```diff
@@ -130,7 +130,7 @@ public static Expiry UnixMilliseconds(Long unixMilliseconds) {
     /** Types of value expiration configuration. */
     @RequiredArgsConstructor
     protected enum ExpiryType {
-        KEEP_EXISTING("KEEPTTL"),
+        KEEP_EXISTING("KEEPTtl"),
```
Comment (jonathanl-bq) [inline java/client/src/main/java/glide/api/models/commands/SetOptions.java:133]: This looks a bit weird and inconsistent with the other entries here now.
